File: dataset_reduzido/avatar/Python/Code/atcoder_AGC033_B.py
H , W , N = map ( int , input ( ).split ( ) )
s_r , s_c = map ( int , input ( ).split ( ) )
S = input ( )
T = input ( )
Judge = False
S_L , S_R , S_U , S_D = 0 , 0 , 0 , 0
T_L , T_R , T_U , T_D = 0 , 0 , 0 , 0
for x in range ( N ) :
    if S [ x ] == 'L' :
        S_L += 1
        if S_L - T_R - s_c == 0 :
            Judge = True
    elif S [ x ] == 'R' :
        S_R += 1
        if s_c + ( S_R - T_L ) == W + 1 :
            Judge = True
    elif S [ x ] == 'U' :
        S_U += 1
        if S_U - T_D - s_r == 0 :
            Judge = True
    elif S [ x ] == 'D' :
        S_D += 1
        if s_r + ( S_D - T_U ) == H + 1 :
            Judge = True
    if T [ x ] == 'L' :
        if S_R - T_L + s_c != 1 :
            T_L += 1
    if T [ x ] == 'R' :
        if s_c + ( T_R - S_L ) != W :
            T_R += 1
    if T [ x ] == 'U' :
        if S_D - T_U + s_r != 1 :
            T_U += 1
    if T [ x ] == 'D' :
        if s_r + ( T_D - S_U ) != H :
            T_D += 1
if Judge :
    print ( 'NO' )
else :
    print ( 'YES' )

File: output/LLaMa/avatar/Python/Java/atcoder_AGC033_B.java
```java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        String[] input = scanner.nextLine().split(" ");
        int H = Integer.parseInt(input[0]);
        int W = Integer.parseInt(input[1]);
        int N = Integer.parseInt(input[2]);

        int[] sR = new int[N];
        int[] sC = new int[N];
        int sRIndex = 0;
        int sCIndex = 0;

        int[] tR = new int[N];
        int[] tC = new int[N];
        int tRIndex = 0;
        int tCIndex = 0;

        String S = scanner.nextLine();
        String T = scanner.nextLine();

        boolean judge = false;

        int[] sLeft = new int[1];
        int[] sRight = new int[1];
        int[] sUp = new int[1];
        int[] sDown = new int[1];

        int[] tLeft = new int[1];
        int[] tRight = new int[1];
        int[] tUp = new int[1];
        int[] tDown = new int[1];

        for (int i = 0; i < N; i++) {
            if (S.charAt(i) == 'L') {
                sLeft[0] += 1;
                if (sLeft[0] - tRight[0] - sCIndex == 0) {
                    judge = true;
                }
            } else if (S.charAt(i) == 'R') {
                sRight[0] += 1;
                if (sCIndex + (sRight[0] - tLeft[0]) == W + 1) {
                    judge = true;
                }
            } else if (S.charAt(i) == 'U') {
                sUp[0] += 1;
                if (sUp[0] - tDown[0] - sRIndex == 0) {
                    judge = true;
                }
            } else if (S.charAt(i) == 'D') {
                sDown[0] += 1;
                if (sRIndex + (sDown[0] - tUp[0]) == H + 1) {
                    judge = true;
                }
            }

            if (T.charAt(i) == 'L') {
                if (sCIndex